### PR TITLE
SER-206 - refactor health check endpoints

### PR DESCRIPTION
**Pull request by xaithe**
## Description

Update endpoints to be consistent as per:

https://docs.opg.service.justice.gov.uk/documentation/adrs/adr-007.html

- Reused existing endpoints where possible, added DB health check to API
- Added health check command to task definitions
- Created alert for each health check endpoint

## Issues Resolved

[SER-206](https://opgtransform.atlassian.net/browse/SER-206)

[SER-206]: https://opgtransform.atlassian.net/browse/SER-206?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: serve-api/handlers/health.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+)
+
+func DbStatusCheck(db *gorm.DB) (bool, string) {
+	database, err := db.DB()
+	if err != nil {
+		return false, "Internal Error"
+	}
+
+	if err := database.Ping(); err != nil {
+		return false, fmt.Sprintf("Database Connection Error: %s", err)
```
Comment (charlesmmarshall) [inline serve-api/handlers/health.go:16]: hey @xaithe - not sure if the project is using strings intentianlly, so excuse me if im wrong! you can do `fmt.Errorf` which will give you an `error`, or in go 1.20+ there is `errors.Join` to merge together more than one

Comment (xaithe) [reply serve-api/handlers/health.go:16]: The intention here is to have a message to send back to the main app when it hits the health check, so I used a string. The go api isn't actually active though so was more doing this for golang practice 🙂

File: docker-compose.yml [hunk 2]
```diff
@@ -35,7 +35,7 @@ services:
       dockerfile: docker/Dockerfile.dev
     image: 311462405659.dkr.ecr.eu-west-1.amazonaws.com/serve-opg/api:latest
     ports:
-      - 9000:9000
+      - 9090:9090
```
Comment (jamesrwarren) [inline docker-compose.yml:38]: I'd be surprised if you need this. This exposes your external port 9090 to internal port 9090 (on the docker network). As nginx will be routing to internal to the network and same for the app, it shouldn't need to be exposed. It's not a problem, but don't think it's needed :)

File: serve-web/src/Service/Availability/DatabaseAvailability.php [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+{
+    private $em;
+
+    public function __construct(EntityManager $em)
+    {
+        $this->em = $em;
+    }
+
+
+    public function ping()
```
Comment (jamesrwarren) [inline serve-web/src/Service/Availability/DatabaseAvailability.php:17]: awesome stuff. I don't know serve very well and not sure if it's cos all the API stuff hasn't been moved to golang, but if the golang api stuff is acting as the backend then normally you'd have this in the api app but tbh it's prolly that not everything has been moved across?

**Review by jamesrwarren — APPROVED** [review]

File: docker-compose.yml (hunk 2 above)
Comment (xaithe) [reply docker-compose.yml:38]: I did this to deconflict with the app being on 9000 as well locally. The API doesn't actually get deployed yet anyway

Comment (xaithe) [reply docker-compose.yml:38]: Ah just seen your other comment, I think I'll leave it for now as it doesn't break anything

**Pull request merged**